### PR TITLE
Density - pounds per gallon, LinearDensity - kg/m and lb/ft

### PR DESCRIPTION
**Pull request by heytherefred**
I added pounds per gallon to the Density unit type and created a new unit type "LinearDensity" and added kilograms per meter and pounds per foot for that.

### CODE REVIEW

File: UnitsNet/UnitDefinitions/Density.json [hunk 1]
```diff
@@ -161,6 +161,18 @@
           "Abbreviations": [ "g/ml" ]
         }
       ]
+    },
+    {
+      "SingularName": "PoundPerGallon",
```
Comment (gojanpaolo) [inline UnitsNet/UnitDefinitions/Density.json]: We might need to separate this into PoundPerUsGallon and PoundPerImperialGallon (or just rename it to make it more explicit).

See Volume: [ImperialGallon](https://github.com/angularsen/UnitsNet/blob/master/UnitsNet/UnitDefinitions/Volume.json#L187) and [UsGallon](https://github.com/angularsen/UnitsNet/blob/master/UnitsNet/UnitDefinitions/Volume.json#L220)

**Comment by gojanpaolo** [comment]
Thank you for contributing to the library. I'll continue review later. So far, awesome work! :)

File: UnitsNet/UnitDefinitions/Density.json (hunk 1 above)
Comment (heytherefred) [reply UnitsNet/UnitDefinitions/Density.json]: Darn, I can't believe I forgot about that.  My dad used to track & calculate his gas mileage in miles per imperial gallon.

**Comment by gojanpaolo** [comment]
@heytherefred Are you planning to use the built-in parser of the library? If so, then please note that it will throw an exception for the ambiguous abbreviation `"ppg"` for `PoundsPerUSGallon` and `PoundsPerImperialGallon`. (See also https://github.com/angularsen/UnitsNet/issues/390, https://github.com/angularsen/UnitsNet/issues/132). 

@angularsen Are ambiguous abbreviations to be avoided adding to the library? If this is ok then I think this can be merged. :)

**Comment by angularsen** [comment]
Great review @gojanpaolo and good catch!

Yes, we should strive to avoid ambiguous abbreviations for units of the same quantity, so that we that can parse them without getting exception.

We have the following convention in our contribution guideline:
https://github.com/angularsen/UnitsNet/blob/master/CONTRIBUTING.md#units
> Generally we try to name the units as what is the most widely used.
> * Use prefix for country variants, such as ImperialGallon and UsGallon

https://github.com/angularsen/UnitsNet/blob/master/CONTRIBUTING.md#unit-abbreviations
> Use suffixes to distinguish variants of similar units, such as `gal (U.S.)` vs `gal (imp.)` for gallons
> `(U.S.)` for United States
> `(imp.)` for imperial / British units

**Comment by heytherefred** [comment]
Ok, that sounds good.  I'll change the abbreviations to "ppg (U.S.)" and "ppg (imp.)".

**Comment by angularsen** [comment]
Perfect!

@gojanpaolo Assuming you verified the unit conversion test values, this looks good to me.
Please have the honors of merging your first PR and releasing a new nuget (bump the minor version for new features). See: https://github.com/angularsen/UnitsNet/wiki/Collaborators:-Release-New-Nuget

**Comment by gojanpaolo** [comment]
Awesome! Will do :)

**Comment by heytherefred** [comment]
Thanks guys!  This is my first github contribution.

**Comment by gojanpaolo** [comment]
Great work! My first github contribution was on UnitsNet too :))

@angularsen I think I'm not doing the release correctly. I now have 3.90.0 only my fork. Do I need to create a pull request for that?
Also, I didn't bump the Json Serialization library.

**Comment by angularsen** [comment]
@gojanpaolo Ah, yes there are some extra set up to do that is not detailed in the wiki.
I'm not sure exactly how things are set up for you right now, but basically you need to change your local `master` branch to have the official repo's `master` branch as its upstream/remote branch instead of your fork's `master`. You now have push-privileges and creating release commit and tag is fine to push directly without creating PR first.

```
git remote rename origin my  # since you are probably currently pushing to your fork, let's rename this
git remote add origin <HTTPS or SSH url to angularsen/unitsnet repo> # add official repo as origin remote
git checkout master
git branch --set-upstream-to=origin/master  # it would probably be my/master at this point
```

Please note that any new branches you create should still be pushed to your fork (`my` remote) so you create PRs from your fork to merge in significant changes.
```
git checkout -b new-branch master  # new branch, based on master
# Add some commits...
git push my new-branch  # push to your fork, with multiple remotes you need to specify remote
```

Let me know if this makes sense to you and I'll update the wiki.

**Comment by gojanpaolo** [comment]
@angularsen Your instructions did the trick! :) Thank you!

I'll continue the release later/tomorrow. I saved it as draft in the meantime.

**Comment by angularsen** [comment]
Awesome, I'll add some more notes to the wiki.

**Pull request merged**